test/unit/foreman/renderer/scope/macros/kickstart_test.rb

```diff
@@ -0,0 +1,310 @@
+require 'test_helper'
+
+class KickstartTest < ActiveSupport::TestCase
+  setup do
+    os = FactoryBot.create(:for_snapshots_rhel9, :with_provision, :with_associations)
+
+    @host = FactoryBot.create(:host, :managed, :build => true, :operatingsystem => os,
+      :interfaces => [
+        FactoryBot.build(:nic_managed, :primary => true),
+        FactoryBot.build(:nic_managed, :provision => true),
+      ])
+
+    template = OpenStruct.new(
+      name: 'Test',
+      template: 'Test'
+    )
+    source = Foreman::Renderer::Source::Database.new(
+      template
+    )
+    @scope = Class.new(Foreman::Renderer::Scope::Base) do
+      include Foreman::Renderer::Scope::Macros::Kickstart
+    end.send(:new, host: @host, source: source)
+    # HostInfo.stubs(:providers).returns([HostDummyEncInfo])
+  end
+
+  describe '#network' do
+    test 'should return a network line for an interface' do
+      actual = @scope.kickstart_network(
+        @host.managed_interfaces.first,
+        host: @host,
+        use_slaac: false,
+        static: false,
+        static6: false
+      )
+
+      assert_match(/network/, actual)
+    end
+
+    test 'should skip non-managed interfaces' do
+      iface = FactoryBot.build(:nic_base, :primary => true)
```

Comment:
Shouldn't this be actually non-managed?
```suggestion
      iface = FactoryBot.build(:nic_base, :primary => true, :managed => false)
```